lib/hiera/backend.rb

```diff
@@ -233,6 +249,16 @@ def lookup(key, default, scope, order_override, resolution_type)
       def clear!
         @backends = {}
       end
+
+      def qualified_lookup(segments, hash)
+        value = hash
+        segments.each do |segment|
+          break if value.nil? || value == :undefined
```

Comment:
Although not documented there are several spec tests that adds the :undefined value. I changed so that when found, the qualified_lookup method will return nil.